classes/GmailTransport.php

```diff
@@ -81,12 +82,43 @@ private static function base64url(Swift_Mime_SimpleMessage $message) {
      */
     public function send(Swift_Mime_SimpleMessage $message, &$failedRecipients = null) {
         try {
-            $encodedMessage = $this::base64url($message);
+            // Check if the total mail mime size is less than 5 MB
+            if(strlen($message->toString()) < 5 * 1024 * 1024) {
+                $encodedMessage = $this::base64url($message);
 
-            $gmailMessage = new Google_Service_Gmail_Message();
-            $gmailMessage->setRaw($encodedMessage);
+                $gmailMessage = new Google_Service_Gmail_Message();
+                $gmailMessage->setRaw($encodedMessage);
 
-            $gmailMessage = $this->googleAPI->getServiceGmail()->users_messages->send('me', $gmailMessage); // 'me' references the currently authenticated user
+                // Immediately send the mail
+                $gmailMessage = $this->googleAPI->getServiceGmail()->users_messages->send('me', $gmailMessage); // 'me' references the currently authenticated user
+            } else {
+                // Use a resumable upload for large mails
+                $gmailMessage = new Google_Service_Gmail_Message();
+
+                // Set client to deferred mode
+                $this->googleAPI->client->setDefer(true);
+                $gmailMessage = $this->googleAPI->getServiceGmail()->users_messages->send('me', $gmailMessage, ['uploadType' => 'resumable']); // Resumable upload type
```

Comment:
Change the `uploadType` to use the class constant `Google_Http_MediaFileUpload::UPLOAD_RESUMABLE_TYPE` instead of a string.